src/fswAlgorithms/orbitControl/lambertSolver/lambertSolver.cpp

```diff
@@ -0,0 +1,669 @@
+
+        double term1 = pow((this->numberOfRevolutions*M_PI + M_PI)/(8.0*T),(2.0/3.0));
+        double term2 = pow((8.0*T)/(this->numberOfRevolutions*M_PI),(2.0/3.0));
+        x0Sol1 = (term1 - 1.0) / (term1 + 1.0);
+        x0Sol2 = (term2 - 1.0) / (term2 + 1.0);
+    }
+
+    std::array<double, 2> x0 = {x0Sol1, x0Sol2};
+    return x0;
+}
+
+/*! This method computes the non-dimensional time of flight (TOF) for a given x
+    @param x free variable of Lambert's problem that satisfies the given time of flight
+    @param N number of revolutions
+    @param lam lambda parameter that defines the problem geometry
+    @return double
+*/
+double LambertSolver::x2tof(double x, int N, double lam)
+{
+    // if x is close to 1.0 (parabolic case), time-of-flight is computed according to Battin
+    double battin = 0.01;
+    double dist = abs(x - 1.0);
+    double u = 1.0 - pow(x, 2);
+    double y = sqrt(1.0- pow(lam, 2) * u);
+
+    double tof;
+    if (dist < 1e-8){
+        // exact time of flight for parabolic case
+
+        tof = 2.0/3.0*(1.0 - pow(lam,3));
+    }
+    else if (dist < battin){
+        // Use Battin formulation
+
+        double eta = y - lam * x;
+        double S1 = 0.5 * (1.0 - lam - x * eta);
+        double Q = 4.0 / 3.0 * hypergeometricF(S1);
+        tof = (pow(eta,3)*Q + 4.0 * lam * eta) / 2.0 + N * M_PI / (pow(abs(u), 1.5));
+    }
+    else{
+        // Use Lancaster formulation
+
+        double f = (y - lam * x) * sqrt(abs(u));
+        double g = x*y + lam * u;
+        double d;
+        if (u > 0.0){
+            // elliptic case
+            double psi = atan2(f,g);
+            d = psi + N*M_PI;
+        }
+        else{
+            // hyperbolic case
+            d = atanh(f/g);
+        }
+
+        // non-dimensional time-of-flight
+        tof = (d/sqrt(abs(u)) - x + lam * y) / u;
+    }
+
+    return tof;
+}
+
+/*! This method computes the derivatives of the time of flight curve T(x)
+    @param x free variable of Lambert's problem that satisfies the given time of flight
+    @param T requested non-dimensional time-of-flight
+    @param lam lambda parameter that defines the problem geometry
+    @return std::array<double, 3>
+*/
+std::array<double, 3> LambertSolver::dTdx(double x, double T, double lam)
+{
+    double DT; // dT/dx
+    double D2T; // d2T/dx2
+    double D3T; // d3T/dx3
+    if (abs(x - 1.0) < 1e-8){
+        // exact derivative for parabolic case
+        DT = 2.0/5.0*(pow(lam,5) - 1.0);
+        D2T = 0.0;
+        D3T = 0.0;
+    }
+    else{
+        double u = 1.0 - pow(x, 2);
+        double y = sqrt(1.0- pow(lam, 2) * u);
+        DT = 1.0/u*(3.0*T*x - 2.0 + 2.0 * pow(lam, 3) * x / y);
+        D2T = 1.0/u*(3.0*T + 5.0*x*DT + 2.0 * (1. - pow(lam, 2))
+                * pow(lam, 3) / pow(y, 3));
+        D3T = 1.0/u*(7.0*x*D2T + 8.0*DT - 6.0 * (1. - pow(lam, 2))
+                * pow(lam, 5) * x / pow(y, 5));
+    }
+
+    std::array<double, 3> DTs = {DT, D2T, D3T};
```

Comment:
should a frame be labeled with `DTs`?

Reply:
No, DTs includes the derivatives of the time of flight T w.r.t the independent variable x of Lambert's problem.